OpenRA.Game/Map/CellLayer.cs

```diff
@@ -49,12 +49,6 @@ public static CellLayer<T> CreateInstance(Func<MPos, T> initialCellValueFactory,
 			return cellLayer;
 		}
 
-		// Resolve an array index from cell coordinates
-		int Index(CPos cell)
-		{
-			return Index(cell.ToMPos(GridType));
```

Comment:
Can't we do here what you currently do in `ToOffset`? I'm not sure if we want to expose that `ToOffset` logical to everything when it is only used here.

Reply:
I first implemented it there. But then it seemed cleaner to keep all `ToMPos` logic in `CPos` - not spread duplicated code. `ToOffset` also seemed generic enough. I don't mind where.